resource/plugins/unreal/python/loader/finalizers/unreal_asset_loader_finalizer.py

```diff
@@ -0,0 +1,171 @@
+        # Transform to asset path
+        asset_path = unreal_utils.filesystem_asset_path_to_asset_path(
+            asset_filesystem_path
+        )
+
+        # Load the asset in Unreal
+        self.logger.debug(
+            'Result of loading asset "{}" in Unreal editor: {}'.format(
+                asset_path, unreal.EditorAssetLibrary.load_asset(asset_path)
+            )
+        )
+
+        # Restore pipeline asset info
+        assetversion = self.session.query(
+            'AssetVersion where id is "{}"'.format(context_data['version_id'])
+        ).one()
+        ident = str_version(assetversion, by_task=False)
+        metadata = None
+        if 'ftrack-connect-pipeline-unreal' in assetversion['metadata']:
+            metadata = json.loads(
+                assetversion['metadata']['ftrack-connect-pipeline-unreal']
+            )
+            if 'pipeline_asset_info' in metadata:
+                asset_info = metadata['pipeline_asset_info']
+
+                dependency_assetversion = self.session.query(
+                    'AssetVersion where id is "{}"'.format(
+                        asset_info[asset_const.VERSION_ID]
+                    )
+                ).one()
+                dependency_ident = str_version(
+                    dependency_assetversion, by_task=False
+                )
+                self.logger.debug(
+                    'Restoring pipeline asset info for {}: {}'.format(
+                        dependency_ident, asset_info
+                    )
+                )
+                ftrack_object_manager = UnrealFtrackObjectManager(
+                    self.event_manager
+                )
+                ftrack_object_manager.asset_info = asset_info
+                dcc_object = UnrealDccObject()
+                dcc_object.name = (
+                    ftrack_object_manager._generate_dcc_object_name()
+                )
+                if dcc_object.exists():
+                    self.logger.debug(
+                        'Pipeline asset {} already tracked in Unreal, removing!'.format(
+                            dependency_ident
+                        )
+                    )
+                    unreal_utils.delete_ftrack_node(dcc_object.name)
+                # Store asset info
+                dcc_object.create(dcc_object.name)
+                # Have it sync to disk
+                ftrack_object_manager.dcc_object = dcc_object
+            else:
+                # Remove metadata
+                if unreal_utils.conditional_remove_metadata_tag(
+                    asset_path, asset_const.NODE_METADATA_TAG
+                ):
+                    self.logger.debug(
+                        'Removed pipeline asset metadata tag from asset: {}'.format(
+                            asset_path
+                        )
+                    )
+        # Remove snapshot metadata tag carried along - this is not a dependency (yet)
```

Comment:
We are in the loader, so that means that the loaded asset is a dependency of something (A level or any other thing) no?

Reply:
Yes, this should only be removed when opening an asset.